### PR TITLE
[opentelemetry-otlp] adds an example HTTP exporter backed by Hyper

### LINKED ISSUE #1659: [Bug]: unclear how to setup http(hyper) without agent pipeline
### What happened?

```
opentelemetry = "0.22"
opentelemetry_sdk = { version = "0.22", features = ["rt-tokio"] }
tracing-opentelemetry = "0.23"
opentelemetry-jaeger-propagator = "0.1.0"
opentelemetry-jaeger = { version = "0.21", features = ["collector_client", "hyper_collector_client", "rt-tokio"] }
opentelemetry-otlp = { version = "0.15.0", features = ["http-proto"] }
```

these are the enabled features

```
global::set_text_map_propagator(opentelemetry_jaeger_propagator::Propagator::new());

        let otlp_exporter = opentelemetry_otlp::new_exporter()
            .http()
            .with_endpoint(jager_endpoint);
        let tracer =
            opentelemetry_otlp::new_pipeline()
                .tracing()
                .with_exporter(otlp_exporter)
                .with_trace_config(trace::config().with_resource(Resource::new(vec![
                    KeyValue::new("service.name", "clash-rs"),
                ])))
                .install_batch(opentelemetry_sdk::runtime::Tokio)?;

```

these are the setup

problem is

```
failed to setup logging: Exporter otlp encountered the following error(s): no http client, you must select one from features or provide your own implementation
```

how do i active the default hyper client?

ty

### API Version

x

### SDK Version

x

### What Exporters are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by markdingram**
Resolves #1659

Given the Pending Hyper 1.0 upgrade seems to make sense for Hyper 0.14 Client support to be provided as an example. 

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by cijothomas** [comment]
From 6/4/ community call:
Please see if we can use the existing examples, and add feature flags to it to showcase diff. http clients, instead of maintaining full examples for each of them.

**Comment by cijothomas** [comment]
> From 6/4/ community call: Please see if we can use the existing examples, and add feature flags to it to showcase diff. http clients, instead of maintaining full examples for each of them.

@markdingram Would you be willing to change this based on this recommendation? Thanks in advance!

**Comment by markdingram** [comment]
> > From 6/4/ community call: Please see if we can use the existing examples, and add feature flags to it to showcase diff. http clients, instead of maintaining full examples for each of them.
> 
> @markdingram Would you be willing to change this based on this recommendation? Thanks in advance!

I had a go

File: opentelemetry-otlp/CHANGELOG.md
Comment (cijothomas) [inline opentelemetry-otlp/CHANGELOG.md:19]: dont need a changelog entry for this.

File: opentelemetry-otlp/examples/basic-otlp-http/README.md
Comment (cijothomas) [inline opentelemetry-otlp/examples/basic-otlp-http/README.md:56]:
```suggestion
By default the app will use a `reqwest` client to send. A hyper client can be used with the `hyper` feature enabled
```

**Review by cijothomas — APPROVED** [review]
Thanks!
Will wait for one more approval to merge. (My own experience with hyper is limited, so need an extra pair of eyes)

File: opentelemetry-otlp/examples/basic-otlp-http/src/hyper.rs
Comment (lalitb) [inline opentelemetry-otlp/examples/basic-otlp-http/src/hyper.rs:1]: Do you think we can instead update the [HttpConfig ](https://github.com/open-telemetry/opentelemetry-rust/blob/e0fb7fe3b7ab343cdf2f702b04d04976dc5542ca/opentelemetry-otlp/src/exporter/http/mod.rs#L53) in opentelemetry-otlp with the hyper client, and so not have the special handling here

**Review by lalitb — CHANGES_REQUESTED** [review]
@markdingram please see if this makes sense - https://github.com/open-telemetry/opentelemetry-rust/pull/1861#discussion_r1646709516

**Comment by markdingram** [comment]
> @markdingram please see if this makes sense - [#1861 (comment)](https://github.com/open-telemetry/opentelemetry-rust/pull/1861#discussion_r1646709516)

Anything done now will conflict with https://github.com/open-telemetry/opentelemetry-rust/pull/1674 - I will look again once the rate upgrades are through.

**Comment by lalitb** [comment]
> > @markdingram please see if this makes sense - [#1861 (comment)](https://github.com/open-telemetry/opentelemetry-rust/pull/1861#discussion_r1646709516)
> 
> Anything done now will conflict with #1674 - I will look again once the crate upgrades are through.

should be good, assuming we will revisit this once #1674 is through.

**Review by lalitb — APPROVED** [review]

**Pull request merged**
